pinot-core/src/main/java/org/apache/pinot/core/data/table/ConcurrentIndexedTableOptimizedForLargeGroups.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.table;
+
+import com.google.common.base.Preconditions;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.core.query.request.context.QueryContext;
+
+
+public class ConcurrentIndexedTableOptimizedForLargeGroups extends ConcurrentIndexedTable {
+
+  public ConcurrentIndexedTableOptimizedForLargeGroups(DataSchema dataSchema,
+      QueryContext queryContext, int trimSize, int trimThreshold) {
+    super(dataSchema, queryContext, trimSize, trimThreshold);
+  }
+
+  @Override
+  public boolean upsert(Key key, Record newRecord) {
+    Preconditions.checkNotNull(key, "Cannot upsert record with null keys");
```

Comment:
Could be skipped for perf?

Reply:
I actually didn't try it to see the improvement. May be we can remove it.